goal_src/levels/beach/seagull.gc

```diff
@@ -5,3 +5,2161 @@
+            )
+           )
+          )
+         0
+         )
+        )
+       (let
+        ((f0-33 (+ (* (-> s4-0 x) (-> s4-0 x)) (* (-> s4-0 z) (-> s4-0 z)))))
+        (set! (-> s5-0 angletan) (/ (- (-> s4-0 y)) (sqrtf f0-33)))
+        (set!
+         (-> s5-0 target-dist)
+         (sqrtf (+ f0-33 (* (-> s4-0 y) (-> s4-0 y))))
+         )
+        )
+       )
+      0
+      (let ((v1-42 (>= (ja-frame-num 0) (ja-aframe 3.0 0))))
+       (cond
+        ((and v1-42 (>= (ja-aframe 9.0 0) (ja-frame-num 0)))
+         (let* ((v1-44 self)
+                (a0-50 #t)
+                (f0-38 (-> v1-44 root-override transv y))
+                )
+          (set! f0-38 (cond
+                       (a0-50
+                        (if (< f0-38 (-> v1-44 flock 0 max-lift))
+                         (set! f0-38 (+ 12288.0 f0-38))
+                         )
+                        f0-38
+                        )
+                       (else
+                        (+ -4096.0 f0-38)
+                        )
+                       )
+           )
+          (set! (-> v1-44 root-override transv y) f0-38)
+          )
+         0
+         )
+        (else
+         (let* ((v1-48 self)
+                (a0-56 #f)
+                (f0-39 (-> v1-48 root-override transv y))
+                )
+          (set! f0-39 (cond
+                       (a0-56
+                        (if (< f0-39 (-> v1-48 flock 0 max-lift))
+                         (set! f0-39 (+ 12288.0 f0-39))
+                         )
+                        f0-39
+                        )
+                       (else
+                        (+ -4096.0 f0-39)
+                        )
+                       )
+           )
+          (set! (-> v1-48 root-override transv y) f0-39)
+          )
+         0
+         )
+        )
+       )
+      (let* ((s5-1 self)
+             (f28-0 (-> self thrust))
+             (f30-2 (* f28-0 (sin (-> s5-1 heading))))
+             (f0-44 (* f28-0 (cos (-> s5-1 heading))))
+             )
+       (set!
+        (-> s5-1 root-override transv x)
+        (+ (* 0.8 (-> s5-1 root-override transv x)) (* 0.2 f30-2))
+        )
+       (set!
+        (-> s5-1 root-override transv z)
+        (+ (* 0.8 (-> s5-1 root-override transv z)) (* 0.2 f0-44))
+        )
+       )
+      0
+      (dummy-27 self)
+      (suspend)
+      (let ((a0-65 (-> self skel root-channel 0)))
+       (set!
+        (-> a0-65 param 0)
+        (the float (+ (-> a0-65 frame-group data 0 length) -1))
+        )
+       (set! (-> a0-65 param 1) 1.0)
+       (joint-control-channel-group-eval!
+        a0-65
+        (the-as art-joint-anim #f)
+        num-func-seek!
+        )
+       )
+      (set! t9-12 ja-done?)
+      (set! a0-66 0)
+      )
+     (+! gp-0 1)
+     (when (>= gp-0 0)
+      (when (>= (-> self angletan) 0.1)
+       (if (< 204800.0 (-> self root-override trans y))
+        (go seagull-soaring (the-as seagull a0-66))
```

Comment:
same here. The `a0-66` should go away and the `ja-done?` a few lines above should hopefully move all the way up to the `until` at the top